### PR TITLE
CI: fix running PT 1.11

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

seems docker was failing for missing `torchmetrics`, but in fact for failing `horovod` build

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @tchaton @rohitgr7 @akihironitta @carmocca @borda

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Comment by Borda** [comment]
@carmocca any idea for the failing `mypy` while I have not touched any code?

File: dockers/base-conda/Dockerfile [hunk 1]
```diff
@@ -59,8 +59,8 @@ ENV \
     LD_LIBRARY_PATH="/root/miniconda3/lib:$LD_LIBRARY_PATH" \
     CUDA_TOOLKIT_ROOT_DIR="/usr/local/cuda" \
     MKL_THREADING_LAYER=GNU \
-    MAKEFLAGS="-j$(nproc)" \
-    # MAKEFLAGS="-j1" \
+    # MAKEFLAGS="-j$(nproc)" \
```
Comment (carmocca) [inline dockers/base-conda/Dockerfile:62]:
```suggestion
```

File: .github/workflows/ci_dockers.yml [hunk 2]
```diff
@@ -95,24 +95,23 @@ jobs:
...
       matrix:
-        # the config used in '.github/workflows/ci_test-conda.yml'
-        python_version: ["3.8"]
-        pytorch_version: ["1.8", "1.9", "1.10"]
+        include:
+          # see: https://pytorch.org/get-started/previous-versions/
+          - {python_version: "3.8", pytorch_version: "1.8", cuda_version: "11.1"}
+          - {python_version: "3.8", pytorch_version: "1.9", cuda_version: "11.1"}
+          - {python_version: "3.8", pytorch_version: "1.10", cuda_version: "11.1"}
+          - {python_version: "3.9", pytorch_version: "1.11", cuda_version: "11.3.1"}
```
Comment (carmocca) [inline .github/workflows/ci_dockers.yml:103]: This PR can be merged but the 1.11 job is failing

Comment (Borda) [reply .github/workflows/ci_dockers.yml:103]: yes, but that will be fixed by creating this new docker image...

File: dockers/base-conda/Dockerfile (hunk 1 above)
Comment (Borda) [reply dockers/base-conda/Dockerfile:62]: would be thinking to make it as argument as seems for GHA 2 is max to not crash the process but locally I want to use all

**Pull request merged**